test/UnitWOPIHttpRedirectLoop.cpp

```diff
@@ -0,0 +1,105 @@
+/* -*- Mode: C++; tab-width: 4; indent-tabs-mode: nil; c-basic-offset: 4; fill-column: 100 -*- */
+/*
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+
+#include <config.h>
+
+#include "WopiTestServer.hpp"
+#include <Log.hpp>
+#include <Unit.hpp>
+#include <UnitHTTP.hpp>
+#include <helpers.hpp>
+
+#include <Poco/Net/HTTPRequest.h>
+
+class UnitWopiHttpRedirectLoop : public WopiTestServer
+{
+    enum class Phase
+    {
+        Load,
+        Redirected,
+        Polling
+    } _phase;
+
+    const std::string params = "access_token=anything";
+    static constexpr auto testname = "UnitWopiHttpRedirectLoop";
+
+public:
+    UnitWopiHttpRedirectLoop()
+        : WopiTestServer(testname)
+        , _phase(Phase::Load)
+    {
+    }
+
+    virtual bool handleHttpRequest(const Poco::Net::HTTPRequest& request,
+                                   Poco::MemoryInputStream& /*message*/,
+                                   std::shared_ptr<StreamSocket>& socket) override
+    {
+        Poco::URI uriReq(request.getURI());
+        Poco::RegularExpression regInfo("/wopi/files/[0-9]+");
+        std::string redirectUri = "/wopi/files/";
+        static unsigned fileId = 1;
+
+        LOG_INF("Fake wopi host request URI [" << uriReq.toString() << "]:\n");
+
+        // CheckFileInfo - always returns redirect response
+        if (request.getMethod() == "GET" && regInfo.match(uriReq.getPath()))
+        {
+            LOG_INF("Fake wopi host request, handling CheckFileInfo");
+
+            assertCheckFileInfoRequest(request);
+
+            LOK_ASSERT_MESSAGE("It is expected to stop requesting after 21 redirections", fileId <= 22);
```

Comment:
Yet another reason why magic numbers are bad: this test will either break, or become useless, when the redirctLimit default changes in the code. Why can't we have this as a compile-time const with a proper name and use that both in the code and tests?

Reply:
done